nasa/harmony
# Harmony 1092 - Add exponential backoff retry behavior to service manager container work polling

## Jira Issue ID
1092

## Description
I added the axios-retry library to allow axios to retry a specified number of times with exponential backoff before failing and throwing an error. 

Axios client timeouts [return an error code of ECONNABORTED](https://github.com/axios/axios/blob/e52e4dbb575fc8bd9cb7d2f5306f30ee82b40b4d/test/unit/adapters/http.js#L84) (the issue encountered in this ticket), which is not included as a retryable error in the default retry function provided by axios-retry. So I added a manual check for this condition in the retryCondition function (`error.code === 'ECONNABORTED'`).

I added an offset of 4 to the retryDelay function (`exponentialDelay(retryNumber + exponentialDelayOffest)`) so that the initial retry delay is sufficiently large (~1.5 seconds).

## Local Test Steps
1. Start services and postgres-localstack without starting Harmony
2. Check the logs of any manager container and notice that the retry/backoff behavior is being logged
3. Start Harmony (now the managers should start getting successful poll responses)
4. Run a successful request
5. Stop Harmony
6. Check the logs of any manager container and notice that the retry/backoff behavior has resumed

## PR Acceptance Checklist
* ✅ Acceptance criteria met
* ✅ Tests added/updated (if needed) and passing
* ❌ Documentation updated (if needed)
